src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -123,7 +124,36 @@ public override bool Execute()
                 if (string.IsNullOrEmpty(request.NuspecFile))
                 {
                     packageBuilder = logic.GetPackageBuilder(request);
+                    string noWarnProperties = null;
+
+                    if (packageBuilder?.Properties?.TryGetValue("NoWarn", out noWarnProperties) == true
+                        && !string.IsNullOrWhiteSpace(noWarnProperties))
+                    {
+                        PackageSpecificWarningProperties packageSpecificWarningProperties = new();
+
+                        foreach (string packageNoWarnProperty in noWarnProperties.Split(new string[] { Environment.NewLine },
+                            StringSplitOptions.RemoveEmptyEntries))
+                        {
+                            string[] packageNoWarnPropertyPars = packageNoWarnProperty.Split(';');
+                            string nugetFramework = packageNoWarnPropertyPars[0];
+                            string packageId = packageNoWarnPropertyPars[1];
+
+                            foreach (string noWarnProperty in packageNoWarnPropertyPars[2].Split(' '))
+                            {
+                                packageSpecificWarningProperties.Add((NuGetLogCode)Enum.Parse(typeof(NuGetLogCode), noWarnProperty), packageId, NuGetFramework.Parse(nugetFramework));
+                            }
+                        }
+
+                        var warningPropertiesCollection = new WarningPropertiesCollection(
+                            null,
+                            packageSpecificWarningProperties,
+                            null);
+
+                        // Override logger with project no warn properties
+                        packArgs.Logger = new PackCollectorLogger(request.Logger, packArgs.WarningProperties, warningPropertiesCollection);
```

Comment:
There's another place where the PackCollectorLogger is created. 

Isn't it a bit weird that the logger is creating in 2 diferent places? 

Makes it easy to make a mistake.